src/main/groovy/de/dkfz/roddy/tools/versions/Version.groovy

```diff
@@ -0,0 +1,166 @@
+    final Integer patch
+    final Integer revision
+
+    public File buildVersionFile = null
+
+    Version (Integer major, Integer minor, Integer patch, Integer revision = 0) {
+        this.major = major
+        this.minor = minor
+        this.patch = patch
+        this.revision = revision
+    }
+
+    Version increaseMajor() {
+        return new Version(major + 1, 0, 0, 0)
+    }
+
+    Version increaseMinor() {
+        return new Version(major, minor + 1, 0, 0)
+    }
+
+    Version increasePatch() {
+        return new Version(major, minor, patch + 1, 0)
+    }
+
+    Version increaseRevision() {
+        return new Version(major, minor, patch, revision + 1)
+    }
+
+    String toString() {
+        return "${major}.${minor}.${patch}-${revision}"
+    }
+
+    private static final Pattern versionPattern = Pattern.compile(/^(\d+)\.(\d+)\.(\d+)(-(\d+))?$/)
+
+    static Version fromString (String versionString) {
+        Matcher matcher = versionPattern.matcher(versionString)
+        if (matcher.find()) {
+            return new Version (
+                    matcher.group(1).toInteger(),
+                    matcher.group(2).toInteger(),
+                    matcher.group(3).toInteger(),
+                    matcher.group(5).toInteger(),
+            )
+        } else {
+            throw new ParseException("Could not parse version string '${versionString}'", 0)
+        }
+    }
+
+
+    @CompileDynamic
+    Integer getAt (int level) {
+        return getAt(VersionLevel.fromInteger(level))
+    }
+
+    @CompileDynamic
+    Integer getAt (VersionLevel level) {
+        if (level == VersionLevel.MAJOR) major
+        else if (level == VersionLevel.MINOR) minor
+        else if (level == VersionLevel.PATCH) patch
+        else if (level == VersionLevel.REVISION) revision
+        else throw new IndexOutOfBoundsException("1.2.3-4")
+    }
+
+    Boolean equals (Version other) {
+        return this.compareTo(other) == 0
+    }
+
+    @Override
+    int compareTo(Version o) {
+        return compareTo(o, VersionLevel.REVISION)
+    }
+
+    /** A more generic version of comparison function that allows to compare up to a specific version level. */
+    int compareTo(Version o, VersionLevel level) {
+        for (l in VersionLevel.MAJOR..level) {
+            int result = this[l].compareTo(o[l])
+            if (result != 0)
+                return result
+        }
+        return 0
+    }
+
+
+    /** If two versions are compatible, then all intermediate versions are also compatible. Compatibility generates an
+     *  equivalence class of versions (reflexive, symmetric, transitive). */
+
+    /** Two versions are compatible, if they have they only differ in the revision, or if the versions are explicitly
+     *  marked as being compatible. The explicit marking of revisions as compatible can be done with the
+     *  compatibilities parameter. If there is any interval that contains both versions (this and the other) then
```

Comment:
When you look at Roddy, a plugin can also be marked as compatible to previous versions of a plugin without being a revision! This happens, when code or functionality is added but the former code stays the same (except bufixes)